lib/serverengine/supervisor.rb

```diff
@@ -194,20 +239,29 @@ def try_join
     end
 
     def start_server
-      s = create_server(logger)
-      @last_start_time = Time.now
+      unless ServerEngine.windows?
+        s = create_server(logger)
+        @last_start_time = Time.now
 
-      begin
-        m = @pm.fork do
-          $0 = @server_process_name if @server_process_name
-          s.install_signal_handlers
+        begin
+          m = @pm.fork do
+            $0 = @server_process_name if @server_process_name
+            s.install_signal_handlers
+
+            s.main
+          end
 
-          s.main
+          return m
+        ensure
+          s.after_start
         end
+      else # if ServerEngine.windows?
+        inpipe, @command_pipe = IO.pipe
+        @last_start_time = Time.now
+        m = @pm.spawn(*Array(config[:windows_daemon_cmdline]), in: inpipe)
```

Comment:
>
* On Windows, if supervisor is enabled and worker type is multi-process stdin of server/worker processes are always command_pipe even if command_sender is "pipe".
* On the other platforms, stdin of server/worker processes will be command_pipe if command_sender is "pipe", inherit parent process if command_sender is "signal".

You're right.
I have to rethink about this inconsistency...